src/components/cc-domain-management/cc-domain-management.js

```diff
@@ -106,99 +116,77 @@ export class CcDomainManagement extends LitElement {
         suggestedElement.focus();
       }
       else {
-        this._emptyMessageRef?.value?.focus();
+        this._emptyMessageRef.value?.focus();
       }
     });
 
     new LostFocusController(this, '.mark-primary', () => {
-      /** @type {HTMLDivElement|null} */
-      const primaryDomainCard = this.shadowRoot?.querySelector('.primary');
-      primaryDomainCard?.focus();
+      focusBySelector(this, '.primary');
     });
   }
 
   /**
-   * @param {string} domainWithPathPrefix
-   * @param {boolean} isHttpOnly
-   * @returns {string}
-   */
-  _getDomainUrl (domainWithPathPrefix, isHttpOnly) {
-    const escapedDomainWithPathPrefix = domainWithPathPrefix.replace('*.', 'www.');
-    return isHttpOnly ? 'http://' + escapedDomainWithPathPrefix : 'https://' + escapedDomainWithPathPrefix;
-  }
-
-  /**
-   * @param {string} domain
+   * @param {string} hostname
    * @returns {FormError|null} The error code or `null` if no error
    */
-  _getErrorCode (domain) {
-    if (domain.length === 0) {
-      return 'empty';
-    }
+  _getErrorCode (hostname) {
```

Comment:
Dicussed in Sync with Pierre (thanks again :heart:)
The `parseDomain` helper throws errors when the domain to parse:
- is empty, 
- has invalid wildcard (`toto*.example.com` is invalid, "*" may only be used as a subdomain, it cannot be part of a subdomain)
- has invalid format.

This is why we use try / catch here.

The `hostname-contains-path` is not part of what the `parseDomain` checks because it's very specific. 
It's only a problem within this component and only when users try to enter a path within the domain input field. In other contexts, domains containing a path are fine since it's a feature Clever Cloud supports.